client/my-sites/term-tree-selector/terms.jsx

```diff
@@ -0,0 +1,413 @@
+	},
+
+	isRowLoaded( { index } ) {
+		return this.props.lastPage || !! this.getItem( index );
+	},
+
+	getItemHeight( item ) {
+		if ( item && this.itemHeights[ item.ID ] ) {
+			return this.itemHeights[ item.ID ];
+		}
+
+		let height = ITEM_HEIGHT;
+
+		if ( item && item.items ) {
+			height += item.items.reduce( ( memo, nestedItem ) => {
+				return memo + this.getItemHeight( nestedItem );
+			}, 0 );
+		}
+
+		return height;
+	},
+
+	getRowHeight( { index } ) {
+		return this.getItemHeight( this.getItem( index ) );
+	},
+
+	getCompactContainerHeight() {
+		return range( 0, this.getRowCount() ).reduce( ( memo, index ) => {
+			return memo + this.getRowHeight( { index } );
+		}, 0 );
+	},
+
+	getResultsWidth() {
+		const { selectorRef } = this.state;
+		if ( selectorRef ) {
+			return selectorRef.clientWidth;
+		}
+
+		return 0;
+	},
+
+	getRowCount() {
+		let count = 0;
+
+		if ( this.props.termsHierarchy ) {
+			count += this.props.termsHierarchy.length;
+		}
+
+		return count;
+	},
+
+	onSearch( event ) {
+		const searchTerm = event.target.value;
+		if ( this.state.searchTerm && ! searchTerm ) {
+			this.props.onSearch( '' );
+		}
+
+		if ( searchTerm === this.state.searchTerm ) {
+			return;
+		}
+
+		if ( ! this.hasPerformedSearch ) {
+			this.hasPerformedSearch = true;
+			analytics.ga.recordEvent( this.props.analyticsPrefix, 'Performed Term Search' );
+		}
+
+		this.setState( { searchTerm } );
+		this.debouncedSearch();
+	},
+
+	getSelectedIds( selected ) {
+		const selectedObjects = selected || this.props.selected;
+		return selectedObjects.map( function( item ) {
+			if ( ! item.ID ) {
+				return item;
+			}
+
+			return item.ID;
+		} );
+	},
+
+	setVirtualScrollRef( ref ) {
+		this.virtualScroll = ref;
+	},
+
+	renderItem( item ) {
+		const onChange = ( ...args ) => this.props.onChange( item, ...args );
+		const setItemRef = ( ...args ) => this.setItemRef( item, ...args );
+
+		const { multiple, defaultTermId, translate } = this.props;
+		const selectedIds = this.getSelectedIds();
+		const itemId = item.ID;
+		const name = unescapeString( item.name ) || translate( 'Untitled' );
+		const checked = includes( selectedIds, itemId );
+		const inputType = multiple ? 'checkbox' : 'radio';
+		const disabled = (
+			multiple &&
+			checked &&
+			defaultTermId &&
+			1 === selectedIds.length &&
+			defaultTermId === itemId
+		);
+
+		const input = (
+			<input
+				type={ inputType }
+				value={ itemId }
+				onChange={ onChange }
+				disabled={ disabled }
+				checked={ checked }
+			/>
+		);
+
+		return (
+			<div
+				key={ itemId }
+				ref={ setItemRef }
+				className="term-tree-selector__list-item">
+				<label>
+					{ input }
+					<span className="term-tree-selector__label">{ name }</span>
+				</label>
+				{ item.items && (
+					<div className="term-tree-selector__nested-list">
+						{ item.items.map( this.renderItem ) }
+					</div>
+				) }
+			</div>
+		);
+	},
+
+	renderNoResults() {
```

Comment:
Using `noRowsRendered` seems pretty clever, and really helps simplify the problem of managing the placeholder as being considered in the row count. Only downside I can think of is that we lose the option to show a placeholder when "refreshing" a search term, e.g. search "Foo", clear, then search "Foo" again. Previous search results will be shown and we load latest results in the background, but no indication we're doing so. Not sure this is a blocker.